tests/test_sbi.py

```diff
@@ -127,7 +138,65 @@ def simulator(params):
         x_obs=x[ind], theta_fid=theta[ind],
         x=x, theta=theta
     )
-
+    
+    # repeat but save to file and do tarp without bootstrapping
+    metric = PosteriorCoverage(
+        backend='sbi', output_path=Path('./toy'), num_samples=nsamples,
+        sample_method='direct', labels=[f'$\\theta_{i}$' for i in range(3)],
+        plot_list=["tarp", "predictions", "coverage", "histogram", "logprob"],
+        save_samples=True,
+    )
+    fig = metric(
+        posterior=posterior,
+        x_obs=x[ind], theta_fid=theta[ind],
+        x=x, theta=theta, bootstrap=False,
+    )
+    
+    # get samples using emcee with the PosteriorSamples class
+    nsamp = 2
+    nchain = 6
+    ntest = 1
+    metric = PosteriorSamples(backend='sbi', output_path=None,
+        num_samples=nsamp, sample_method='emcee', 
+        labels=[f'$\\theta_{i}$' for i in range(3)],
+        sample_params={'num_chains':nchain},
+     )
+    samples = metric(
+        posterior=posterior,
+        x_obs=x[ind], theta_fid=theta[ind],
+        x=x[:ntest], theta=theta[:ntest,:],
+        skip_initial_state_check=True,
+    )
+    unittest.TestCase().assertIsInstance(samples, np.ndarray)
+    unittest.TestCase().assertListEqual(list(samples.shape), [nsamp*nchain,ntest,3])
+    
+    # get samples using direct method with PosteriorSamples class
+    metric = PosteriorSamples(backend='sbi', output_path=Path('./toy'),
+        num_samples=nsamp, sample_method='direct', 
+        labels=[f'$\\theta_{i}$' for i in range(3)],
+     )
+    samples = metric(
+        posterior=posterior,
+        x_obs=x[ind], theta_fid=theta[ind],
+        x=x[:ntest], theta=theta[:ntest,:],
+    )
+    unittest.TestCase().assertIsInstance(samples, np.ndarray)
+    unittest.TestCase().assertListEqual(list(samples.shape), [nsamp,ntest,3])
+    
+    # get samples using pyro with the PosteriorSamples class
+    metric = PosteriorSamples(backend='sbi', output_path=None,
+        num_samples=nsamp, sample_method='slice_np_vectorized', 
+        labels=[f'$\\theta_{i}$' for i in range(3)],
+        sample_params={'num_chains':nchain},
+     )
+    # samples = metric(
```

Comment:
Commented on purpose?

Reply:
Again, this should have been uncommented, so I have done this